atlas/readme.md

```diff
@@ -0,0 +1,325 @@
+    - all fields explicitly declared by the type or type extension
+    - all names introduced with `let`
+2. The **schema namespace**, containing:
+    - all types explicitly declared in the schema file
+    - all names introduced with `let`
+3. For each ancestor directory up to the [atlas root](#atlas-root) or filesystem root, the namespace of that directory, containing:
+    - the base names (extension stripped) of all files with recognized schema extensions: `.star`, `.gql`, and `.graphql`. It is an error for two sibling schemas to share a base name (i.e. you cannot have `users.star` and `users.gql` as siblings in the same directory); the compiler will warn about this situation and refuse to resolve the name entirely.
+    - the names of any directories containing, in themselves or their 
+4. If the root is a child, the scope chain continues up the directory tree to the next atlas root.
+    
+    
+### atlas root
+
+A `.atlas` directory marks the root of a constellation analagously to how a `.git` directory marks the root of a git repository.
+
+An empty `.atlas` directory is a valid root.
+
+Specifically, an `.atlas` directory contains an `atlas.toml` file, which takes [this form](./.atlas/atlas.toml):
+
+```toml
+# should atlas lookup proceed up the directory structure?
+#   default: false
+child = false
+
+[atlases.main]
+  id = "31498eaba8eb44d5a4ecc4acf7471469"
+  url = "https://atlas.apollo.dev"
+
+[atlases.internal]
+  id = "a223bbc2347dfaefsdf3fefdssdac382"
+  url = "https://some-internal-atlas.somewhere"
+```
+
+This `atlas.toml` defines two atlases: `main` and `internal`.
+
+`main` is the default atlas. There’s nothing special about main atlas, really, except that it’s the atlas that’s used if no atlas is specified.
+
+`.atlas` directories may have a cache. If present, the cache is treated as read only unless explicitly updated.
+
+### global atlas
+
+The global atlas lives in `~/.atlas`. It has the same structure as any other `.atlas` folder, with the exception that the cache is always present.
+
+### atlas cache
+
+TK, but generally we use cacache holding the current state for each atlas under its proper uri as a key. The state is a lemma which references others, also held in the cache under their own hash.
+
+## Distributed data format
+
+We can represent an atlas as a collection of signed structs. These are sketched in [data.star](./data.star).
+
+The first such struct an *axiom*. It's nothing more than a message describing the atlas and signed by some identity. The hash of the entire axiom becomes the atlas' id.
+
+From there, we can change the atlas by creating *lemmas*. A lemma is a signed message describing some change to the atlas. For example, lemmas  can `Grant` other identities write access to namespaces within the atlas, or `Revoke` the same. We can also `Announce` endpoints for schemas, create lemmas for directive metadata, and so on.
```

Comment:
I think we need to understand better the signing process here. Can you explain the why and how of that?